fbwikka: custom handler for 'rawlist'

jayrm opened the pull request:
This custom handler provides a list of pages (index) in a raw format, free from any HTML formatting.

Purpose is to provide an efficient method for fbdoc tools to obtain a page list from the wiki.

- supported by PageIndex and RecentChanges wiki pages only
- the handler outputs the list of wiki page names in a plain format

Proposed usage:
`https://www.freebasic.net/wiki/wikka.php?wakka=PageIndex/rawlist`
`https://www.freebasic.net/wiki/wikka.php?wakka=RecentChanges/rawlist`

Currently, FreeBASIC's fbdoc tool `getindex` obtains an index of all the wiki pages by fetching the wiki's PageIndex page in HTML format.  The HTML formatted download is more than 300 KB.  After stripping the HTML tags to extract the page names, the actual data is less than 20KB.

----
A useful extension to this may be:
- Also write the page ID (which increments every time the page is changed) with each page name
- by comparing a locally saved index with a newly fetched index, it would be possible to determine which pages have changed, and then fetch only those pages.  A full download ( 3 MB+) of all pages from the wiki takes quite a while and stresses the server needlessly if it can be easily determined only a few pages need to be updated locally.

jayrm committed:
fbwikka: custom handler for 'rawlist'

- supported by PageIndex and RecentChanges wiki pages only
- the handler outputs the list of pages in a plain format

dkl commented on handlers/rawlist/rawlist.php:
```diff
@@ -0,0 +1,47 @@
+	{
+		if( $pages = $this->LoadPageTitles() )
+		{
+			foreach ($pages as $page)
```
The two foreach loops look equal, so it might be better to refactor them into one function.

jayrm replied:
Duplicate code has been removed in a later commit.

dkl reviewed (commented):
Sounds like a useful extension!

jayrm committed:
fbwikka: custom handler for 'rawlist'

- add header to the output for meta information
- add ID field to index
- refactor duplicate code

jayrm commented:
In last commit, I added the `ID` field.  I also added a header so it is possible to determine (reasonably guess) at the format of the file and the validity; or make changes to format in future.

Here is a sample of the output from my local server:
```
# freebasic wiki index
# index: PageIndex
# source: https://myweb.local/fbwikka.git/wikka.php?wakka=
# count: 1075
16635	AdminBadWords
16621	AdminPages
...
...
17824	CatPgCasting
21236	CatPgCompilerSwitches
22442	CatPgCompOpt
17878	CatPgConsole
21597	CatPgControlFlow
21598	CatPgDate
...
```

I think this can help fbdoc & tools.  But, if somebody wanted to just get just the page names without any custom tools, for example fetching the list with `wget`, can still get a plain list of names with something simple like `sed -e '/^#/d' -e 's/^.*\t//' input-file`

jayrm committed:
fbwikka: custom handler for 'rawlist'

- rawlist handler will accept '&format=index' or '&format=list' (default) parameter to specify format.

jayrm commented:
Added possibility for both a plain list and an index that includes ID's:

Proposed usage:
```
https://www.freebasic.net/wiki/wikka.php?wakka=PageIndex/rawlist&format=list
https://www.freebasic.net/wiki/wikka.php?wakka=PageIndex/rawlist&format=index
```

Default is 'list' format if extra parameter not given, or is some unrecognized value.

I feel that this pull request is complete and ready for review.
